### ISSUE TITLE
[Legacy API] Creating a shipment with warehouse_id doesn't work 

### ISSUE THREAD

**Issue by pat154** [ShipEngine/shipengine-js#111]
Hi guys, I'm very eager to start using your new Typescript library! For now I'm using the legacy one and came across the issue below. The owner AustinBratcher suggested that I move the question over to here.

Whenever I attempt to create a shipment using `warehouse_id` I constantly get the error:
```
"errors": [
  {
    "error_source": "shipengine",
    "error_type": "business_rules",
    "error_code": "unspecified",
    "message": "ship_from and warehouse_id cannot be provided in same request",
    "field_name": "warehouse_id",
    "field_value": "1506406"
  }
]
```

Except that I'm not actually including the `ship_from` field in my request. Here's what I'm sending:
```
engine.createShipment([{
  ship_to: {
    city_locality: 'Elkins',
    postal_code: '72727',
    company_name: undefined,
    country_code: 'us',
    state_province: 'AR',
    address_line1: '21934 Raven Rd',
    address_line2: undefined,
    name: 'Mr. Placeholder',
    phone: '(555) 555-555'
  },
  warehouse_id: 'se-1506406',
  packages: [ { weight: [Object], dimensions: [Object] } ]
}]);
```

and somehow the response from shipengine knows about the warehouses address but still hold the `warehouse_id`:
```
{
    "name": "StatusCodeError",
    "statusCode": 400,
    "message": "400 - {\"request_id\":\"30a19229-5b05-4980-b789-435509e71183\",\"errors\":[{\"error_source\":\"shipengine\",\"error_type\":\"business_rules\",\"error_code\":\"unspecified\",\"message\":\"ship_from and warehouse_id cannot be provided in same request\",\"field_name\":\"warehouse_id\",\"field_value\":\"1506406\"}]}",
    "error": {
        "request_id": "30a19229-5b05-4980-b789-435509e71183",
        "errors": [
            {
                "error_source": "shipengine",
                "error_type": "business_rules",
                "error_code": "unspecified",
                "message": "ship_from and warehouse_id cannot be provided in same request",
                "field_name": "warehouse_id",
                "field_value": "1506406"
            }
        ]
    },
    "options": {
        "method": "POST",
        "uri": "https://api.shipengine.com/v1/rates",
        "headers": {
            "Content-type": "application/json",
            "api-key": "[REDACTED]"
        },
        "json": true,
        "body": {
            "shipment": {
                "errors": [],
                "address_validation": null,
                "shipment_id": "se-108536547",
                "carrier_id": null,
                "service_code": null,
                "external_shipment_id": null,
                "ship_date": "2021-05-03T00:00:00Z",
                "created_at": "2021-05-03T02:42:04.3345574Z",
                "modified_at": "2021-05-03T02:42:04.3345574Z",
                "shipment_status": "pending",
                "ship_to": {
                    "name": "Mr. Placeholder",
                    "phone": "(555) 555-555",
                    "company_name": null,
                    "address_line1": "21934 Raven Rd",
                    "address_line2": null,
                    "address_line3": null,
                    "city_locality": "Elkins",
                    "state_province": "AR",
                    "postal_code": "72727",
                    "country_code": "us",
                    "address_residential_indicator": "unknown"
                },
                "ship_from": {
                    "name": "John Doe",
                    "phone": "(254) 773-0061",
                    "company_name": null,
                    "address_line1": "4160 Southerland Rd",
                    "address_line2": null,
                    "address_line3": null,
                    "city_locality": "Troy",
                    "state_province": "TX",
                    "postal_code": "76557",
                    "country_code": "US",
                    "address_residential_indicator": "unknown"
                },
                "warehouse_id": "se-1506406",
                "return_to": {
                    "name": "John Doe",
                    "phone": "(254) 773-0061",
                    "company_name": null,
                    "address_line1": "4160 Southerland Rd",
                    "address_line2": null,
                    "address_line3": null,
                    "city_locality": "Troy",
                    "state_province": "TX",
                    "postal_code": "76557",
                    "country_code": "US",
                    "address_residential_indicator": "unknown"
                },
                "confirmation": "none",
                "customs": {
                    "contents": "merchandise",
                    "customs_items": [],
                    "non_delivery": "return_to_sender"
                },
                "external_order_id": null,
                "order_source_code": null,
                "advanced_options": {
                    "bill_to_account": null,
                    "bill_to_country_code": null,
                    "bill_to_party": null,
                    "bill_to_postal_code": null,
                    "contains_alcohol": false,
                    "delivered_duty_paid": false,
                    "non_machinable": false,
                    "saturday_delivery": false,
                    "dry_ice": false,
                    "dry_ice_weight": null,
                    "freight_class": null,
                    "custom_field1": null,
                    "custom_field2": null,
                    "custom_field3": null,
                    "collect_on_delivery": null
                },
                "insurance_provider": "none",
                "tags": [],
                "packages": [
                    {
                        "package_code": "package",
                        "weight": {
                            "value": 0.2,
                            "unit": "kilogram"
                        },
                        "dimensions": {
                            "unit": "centimeter",
                            "length": 25.8,
                            "width": 67,
                            "height": 53
                        },
                        "insured_value": {
                            "currency": "usd",
                            "amount": 0
                        },
                        "label_messages": {
                            "reference1": null,
                            "reference2": null,
                            "reference3": null
                        },
                        "external_package_id": null
                    }
                ],
                "total_weight": {
                    "value": 0.2,
                    "unit": "kilogram"
                },
                "items": []
            },
            "rate_options": {
                "carrier_ids": [
                    "se-584304",
                    "se-584303"
                ]
            }
        },
        "simple": true,
        "resolveWithFullResponse": false,
        "transform2xxOnly": false
    },
    "response": {
        "statusCode": 400,
        "body": {
            "request_id": "30a19229-5b05-4980-b789-435509e71183",
            "errors": [
                {
                    "error_source": "shipengine",
                    "error_type": "business_rules",
                    "error_code": "unspecified",
                    "message": "ship_from and warehouse_id cannot be provided in same request",
                    "field_name": "warehouse_id",
                    "field_value": "1506406"
                }
            ]
        },
        "headers": {
            "date": "Mon, 03 May 2021 02:42:05 GMT",
            "content-type": "application/json; charset=utf-8",
            "content-length": "359",
            "connection": "close",
            "x-shipengine-requestid": "30a19229-5b05-4980-b789-435509e71183",
            "vary": "Origin",
            "access-control-allow-origin": "https://www.shipengine.com"
        },
        "request": {
            "uri": {
                "protocol": "https:",
                "slashes": true,
                "auth": null,
                "host": "api.shipengine.com",
                "port": 443,
                "hostname": "api.shipengine.com",
                "hash": null,
                "search": null,
                "query": null,
                "pathname": "/v1/rates",
                "path": "/v1/rates",
                "href": "https://api.shipengine.com/v1/rates"
            },
            "method": "POST",
            "headers": {
                "Content-type": "application/json",
                "api-key": "[REDACTED]",
                "accept": "application/json",
                "content-length": 2171
            }
        }
    }
}
```

I'm not sure how, but somehow the addresses are getting into the request along-side the warehouse id.

Any help on this would be much appreciated. Thanks!

**Comment by JamesMessinger** [comment]
Thanks for the issue @pat154.  We're working hard on the TypeScript SDK and hope to have it done next month. It will include the ability to create shipments using a warehouse ID. As soon as it's ready, I'll post here so you can try it out!

**Comment by pierce-h** [comment]
@pat154 last week we published our [1.0 release candidate](https://www.npmjs.com/package/shipengine). It addresses the issue above. I am going to go ahead and close this issue. Thank you for reporting it!
